File: Xaas/Python_이상탐지 알고리즘/standardization_data.py
import numpy as np


def standardization_data(X0, mu0, sds0, numobs, SD_type):
    if SD_type == 'Mean_centered':
        X0std = X0 - np.tile(mu0, (numobs, 1))
    elif SD_type == 'Z-score':
        X0std = (X0 - np.tile(mu0, (numobs, 1))) / np.tile(sds0, (numobs, 1))
    return X0std

File: Xaas/Comparative_study/0_PCA_SS/Standardization_Data.m
function X0std=Standardization_Data(X0,mu0,sds0,numobs,SD_type)
switch SD_type
    case 'Mean_centered'
        X0std = (X0 - repmat(mu0,numobs,1));
    case 'Z-score'
        X0std = (X0 - repmat(mu0,numobs,1)) ./ repmat(sds0,numobs,1);
end

end